### PR TITLE
Fix neglecting next element in scope-entries in visibleSymbols() API

### LINKED ISSUE #27816: Symbols are dropped from the visibleSymbols API
**Description:**
Consider the following source snippet,

```
type TestRec record {|
    
|};

annotation TestRec test1 on function;

# Description
#
# + test - test Parameter Description 
# + boo - boo Parameter Description 
# + return - Return Value Description
public function test1(int test, int boo) returns int {
    <cursor>
    return 12;
}
```

**Steps to reproduce:**
When we get the visible symbols at the given cursor position, only the `test1` annotation is captured and the function `test1` has been dropped.

**Affected Versions:**
SLP8. at least

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
The next element of the scope-entries are not considered when retrieving the visible symbols through `SemanticModel.visibleSymbols()` API

Fixes #27816 

## Approach
Recursively look in the next element of the scope entry and add it to list compiled-symbols if they satisfy the following arguments.
- not null
- not already included in the set of symbols

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by nadeeshaan** [comment]
Shall we add test cases for this?

**Comment by mohanvive** [comment]
@dulajdilshan Shall we handle the review comments and get this merged?

File: language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/config/query_expr_ctx_let_clause_config11.json [hunk 1]
```diff
@@ -418,57 +418,57 @@
...
-      "insertText": "c3",
+      "insertText": "p1",
       "insertTextFormat": "Snippet"
     },
     {
-      "label": "outputNameString",
+      "label": "p2",
       "kind": "Variable",
-      "detail": "string",
+      "detail": "$CompilationError$",
```
Comment (rasika) [inline language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/config/query_expr_ctx_let_clause_config11.json:453]: @nadeeshaan is this expected? `$CompilationError$`

File: language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/config/query_expr_ctx_let_clause_config11.json [hunk 2]
```diff
@@ -516,6 +508,22 @@
...
+      "kind": "Variable",
+      "detail": "$CompilationError$",
+      "sortText": "B",
+      "insertText": "c1",
+      "insertTextFormat": "Snippet"
+    },
+    {
+      "label": "c2",
+      "kind": "Variable",
+      "detail": "$CompilationError$",
```
Comment (rasika) [inline language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/config/query_expr_ctx_let_clause_config11.json:523]: why we are getting `CompilationError` as detail?

File: language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/config/query_expr_ctx_let_clause_config11.json (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/test/resources/completion/new/expression_context/config/query_expr_ctx_let_clause_config11.json:453]: No, this particular issue is an existing one, where the compiler fails to derive the variable's type and then generate it as CompilationError type.

**Review by nadeeshaan — APPROVED** [review]

**Pull request merged**